File: Leetcode - Java/487_array_findMaxConsecutiveOnes.java
// https://leetcode.com/problems/max-consecutive-ones-ii/

class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int pre = 0, curr = 0, result = 0;
        for(int n : nums) {
            if(n  == 1) {
                curr++;
            } else {
                pre = curr + 1;
                curr = 0;
            }
            result = Math.max(result, pre + curr);
        }
        return result; 
    }
}

File: Leetcode-Python-New/487_array_findMaxConsecutiveOnes.py
# https://leetcode.com/problems/max-consecutive-ones-ii/

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        left = maxlen = zeros = 0
        
        for right in range(len(nums)):
            if nums[right] == 0:
                zeros += 1
                
            while zeros == 2:
                if nums[left] == 0:
                    zeros -= 1
                left += 1
            
            maxlen = max(maxlen, right - left + 1)
            
        return maxlen